spring-kafka/src/main/java/org/springframework/kafka/core/StreamsBuilderFactoryBean.java

```diff
@@ -197,22 +209,27 @@ public void stop(Runnable callback) {
 	public synchronized void start() {
 		if (!this.running) {
 			try {
-				Assert.notNull(this.streamsConfig, "'streamsConfig' must not be null");
-				this.kafkaStreams = new KafkaStreams(getObject().build(), this.streamsConfig, this.clientSupplier);
-				this.kafkaStreams.setStateListener(this.stateListener);
-				this.kafkaStreams.setUncaughtExceptionHandler(this.exceptionHandler);
-				if (this.cleanupConfig.cleanupOnStart()) {
-					this.kafkaStreams.cleanUp();
-				}
-				this.kafkaStreams.start();
-				this.running = true;
+				Topology topology = getObject().build();
+				logger.debug(topology.describe());
+				doStart(topology);
 			}
 			catch (Exception e) {
 				throw new KafkaException("Could not start stream: ", e);
 			}
 		}
 	}
 
+	private void doStart(Topology topology) {
```

Comment:
I'm not sure that we need to extract this method. There is no too much complexity and no any exception handling.

Reply:
Adding
```
Topology topology = getObject().build();
logger.debug(topology.describe());
```
increases its complexity a little.

Reply:
Well, that's great that you took my attention to this `logger.debug()`.
To be honest I don't want to have it, at least not without `logger.isDebugEnabled()`.

You just moved 8 lines of code to the method without any value in the future. 
That new `doStart()` method doesn't bring any distribution of responsibility.
That's how I would like to live without it. 
More over it's not going to be called from any other place.

Right, we might think different way for code design, but our style (at least mine) is slightly different and I'm not convinced yet in this new method.

Thank you for understanding